refinery/data_set_manager/tasks.py

```diff
@@ -273,77 +277,71 @@ def parse_isatab(username, public, path, identity_id=None,
-            generate_bam_index(auxiliary_node.file_item.uuid, datafile_path)
-
-        generate_auxiliary_file.update_state(state=celery.states.SUCCESS)
-
-        logger.debug("Auxiliary file for %s generated in %s "
-                     "seconds." % (datafile_path, time.time() - start_time))
-    except Exception as e:
-        logger.error(
-            "Something went wrong while trying to generate the auxiliary file "
-            "for %s. %s" % (datafile_path, e))
-        generate_auxiliary_file.update_state(state=celery.states.FAILURE)
-
-        raise celery.exceptions.Ignore()
-
-
-def generate_bam_index(auxiliary_file_store_item_uuid, datafile_path):
-    """
-    Generate a bam_index file and associate it with the auxiliary
-    FileStoreItem from our generate_auxiliary_file task
-    :param auxiliary_file_store_item_uuid: uuid of FileStoreItem to generate
-    auxiliary file for
-    :type auxiliary_file_store_item_uuid: string
-    :param datafile_path: Full path on disk to the datafile that we want to
-    generate a bam index file for
-    :type datafile_path: string
-    """
-
-    # Try and fetch the bam_index FileExtension
-    # NOTE: that we are not handling the normal errors for an orm.get()s below
-    # because we want the task from which this function is called within to
-    # fail if we can't get what we want.
-    bam_index_file_extension = FileExtension.objects.get(name="bai").name
-    auxiliary_file_store_item = FileStoreItem.objects.get(
-        uuid=auxiliary_file_store_item_uuid)
-
-    # Leverage pysam library to generate bam index file
-    # FIXME: This should be refactored once we don't have a need for
-    # Standalone IGV because this is creating a bam_index file in the same
-    # directory as it's bam file
-    pysam.index(bytes(datafile_path))
-
-    # Map source field of FileStoreItem to path of newly created bam index file
-    auxiliary_file_store_item.source = "{}.{}".format(
-        datafile_path, bam_index_file_extension)
+        if not settings.REFINERY_S3_USER_DATA:
+            datafile_path = parent_node.file_item.datafile.path
+        else:
+            datafile_path = parent_node.file_item.datafile.name
+    except ValueError:
+        logger.error("No datafile for %s", parent_node.file_item)
+        return auxiliary_file_store_item.uuid
+
+    start_time = time.time()
+    logger.debug("Starting auxiliary file gen. for %s" % datafile_path)
+
+    if parent_node.file_item.get_extension().lower() == 'bam':
+        try:
+            auxiliary_file_path = generate_bam_index(datafile_path)
+        except RuntimeError as exc:
+            logger.error("Error while generating auxiliary file for %s: %s",
+                         datafile_path, exc)
+            generate_auxiliary_file.update_state(state=celery.states.FAILURE)
+            raise celery.exceptions.Ignore()
+    else:  # this should never occur
+        auxiliary_file_path = ''
```

Comment:
I think the task should just fail at this point instead.

Reply:
ok